crates/bevy_ecs/src/schedule/parallel_executor.rs

```diff
@@ -204,6 +208,25 @@ impl ExecutorStage {
                 }
             }
 
+            // Verify that dependents are not duplicated
+            #[cfg(debug_assertions)]
+            for system_index in prepare_system_index_range.clone() {
+                let mut system_dependents_set = std::collections::HashSet::new();
+                for dependent_system in &self.system_dependents[system_index] {
+                    let inserted = system_dependents_set.insert(*dependent_system);
+
+                    // This means duplicate values are in the system_dependents list
+                    // This is reproducing when archetypes change. When we fix this, we can remove
+                    // the hack below and make this a debug-only assert or remove it
+                    debug_assert!(inserted);
+                }
+            }
+
+            // Clear the ready events lists associated with each system so we can rebuild them
+            for ready_events_of_dependents in &mut self.ready_events_of_dependents {
```

Comment:
wont this clear all ready events instead of just the ready events of the current prepare_system_index_range? could we move this next to ` self.system_dependents[system_index].clear()` (and the potential `   self.system_dependencies[system_index].clear()`) so all of our "per-system-slice archetype-dependent data resets" happen in the same place?